packages/realm/src/Realm.ts

```diff
@@ -651,8 +662,11 @@ export class Realm {
       if (extras) {
         objectSchema.ctor = extras.constructor;
       }
+      // Some tests require that the properties exist on the object, even
+      // if the value is `undefined`. Thus, explicitly set to `undefined`.
```

Comment:
This comment seems more appropriate for the PR. I do think this is by design - the canonical schema is verbose and therefore includes keys even if the value of those are undefined. We could potentially lift that restriction 🤔

Reply:
Hmm, I don't at this point see an issue with having the fields set to `undefined`. It may be helpful to see all fields in the canonical version? But if we decide to change this at some point, we should consider the behavioral change for people using the `in` operator as `"key1" in mySchema` will go from `true` to `false`.

(I'll remove the comment tho) 👍